saleor/graphql/checkout/mutations/utils.py

```diff
@@ -87,8 +89,25 @@ def clean_delivery_method(
     return method in valid_methods
 
 
-def update_checkout_shipping_method_if_invalid(
+def _is_external_shipping_valid(checkout_info: "CheckoutInfo"):
+    if external_shipping_id := get_external_shipping_id(checkout_info.checkout):
+        for method in checkout_info.valid_delivery_methods:
```

Comment:
Maybe just `return external_shipping_id in [method.id for method in checkout_info.valid_delivery_methods]`, WDYT?

Reply:
merged accidentally but this is a good point, i'm preparing ports, will do so there.